Manually rebased for bb-tinycore.

From d73f1f024a8f179e38ce15dc66938ee40325e5b3 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 3 Apr 2013 08:33:28 +0200
Subject: [PATCH 03/14] beaglebone: HACK: always return 1 for is_bone_lt

This is to make the eMMC flasher work, the EEPROM hasn't been populated yet

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 board/ti/am335x/board.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/board/ti/am335x/board.c b/board/ti/am335x/board.c
index b371376..c92e1e9 100644
--- a/board/ti/am335x/board.c
+++ b/board/ti/am335x/board.c
@@ -459,7 +459,7 @@ void sdram_init(void)
 	if (board_is_evm_sk(&header))
 		config_ddr(303, &ioregs_evmsk, &ddr3_data,
 			   &ddr3_cmd_ctrl_data, &ddr3_emif_reg_data, 0);
-	else if (board_is_bone_lt(&header))
+	else if (board_is_bone_lt(&header) && !board_is_bone(&header))
 		config_ddr(400, &ioregs_bonelt,
 			   &ddr3_beagleblack_data,
 			   &ddr3_beagleblack_cmd_ctrl_data,
diff --git a/board/ti/am335x/board.h b/board/ti/am335x/board.h
index bc700d5..784c288 100644
--- a/board/ti/am335x/board.h
+++ b/board/ti/am335x/board.h
@@ -36,7 +36,8 @@ static inline int board_is_bone(struct am335x_baseboard_id *header)
 
 static inline int board_is_bone_lt(struct am335x_baseboard_id *header)
 {
-	return !strncmp(header->name, "A335BNLT", HDR_NAME_LEN);
+	// return !strncmp(header->name, "A335BNLT", HDR_NAME_LEN);
+	return 1;
 }
 
 static inline int board_is_evm_sk(struct am335x_baseboard_id *header)
